javagen/src/main/java/com/azure/autorest/template/ClientMethodTemplate.java

```diff
@@ -993,4 +1236,20 @@ protected void generateSendRequestSync(ClientMethod clientMethod, JavaType typeB
             function.methodReturn("this.sendRequestAsync(httpRequest).contextWrite(c -> c.putAll(FluxUtil.toReactorContext(context).readOnly())).block()");
         });
     }
+
+    /*
+     * If the ClientMethod doesn't have a parameter for Context the use of 'context' needs to be replaced with
+     * Context.NONE.
+     */
+    private static String replaceContextWithContextNone(ClientMethod clientMethod, String arguments) {
+        if (!clientMethod.getProxyMethod().isSync()) {
+            return arguments;
+        }
+
+        if (!clientMethod.getParameters().contains(ClientMethodParameter.CONTEXT_PARAMETER)) {
+            return arguments.replace("context", "Context.NONE");
```

Comment:
This seems apply to any argument that contains the word "context".

This replacement seems better done on `serviceMethodArgs`?

Reply:
Moved this replace to where it maps client method parameters to proxy method parameters